teps/0048-task-results-without-results.md

```diff
@@ -426,6 +431,100 @@ with the git events such as `when a pull request is created`, etc but it
 doesn't runs. Having default results defined in `check-name-matches` Task
 will fix the `Pipeline`.
 
+## Proposal
+
+We propose adding an optional field - `default` - to the `Result` specification in `Task`.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: task
+spec:
+  results:
+  - name: merge_status
+    description: whether to rebase or squash
+    type: string
+    default: rebase
+  - name: branches
+    description: branches in the repository
+    type: array
+    default:
+    - main
+    - v1alpha1
+    - v1beta1
+    - v1
+  - name: images
+    type: object
+    properties:
+      node:
+       type: string
+       default: "node:latest" 
+      gcloud:
+       type: string
+       default: "gcloud:latest"
+  steps:
+  ...
+```
+
+
+Adding a default value to `Results` is optional; validation of a `Task` doesn't
+fail if the default value isn't provided.
+
+Adding a default value to a `Result` will guarantee that it will hold a value even
+if the `Task` fails to produce it. If a `Task` does not produce a `Result` that does not
+have a default value, then the `Task` should fail - see [tektoncd/pipeline#3497](https://github.com/tektoncd/pipeline/issues/3497) for further details.
+
+## Test Plan
+
+1. Unit tests for the behaviour of `Results` when default value is present and none of the steps produce any result
+1. Unit tests for the behaviour of `Results` when no default value is present and none of the steps produce any result
+1. E2E tests can take the form of YAML examples that demonstrate the syntax for declaring a result with default value and results without defaults
+
+## Alternatives
+
+### Declaring Results as Optional
+
+Allow `Results` to declare an optional field as `optional`. When a `Task` fails to
+produce a `Result` that's optional, then the `Task` does not fail. When a `Task` fails
+to produce a `Result` that's not optional, then then the `Task` will fail.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: task
+spec:
+  results:
+  - name: merge_status
+    description: whether to rebase or squash
+    type: string
+    optional: true
+  - name: branches
+    description: branches in the repository
+    type: array
+    optional: true
+  - name: images
+    type: object
+    properties:
+      node:
+       type: string
+       optional: true 
+      gcloud:
+       type: string
+  steps:
+  ...
+```
+
+However, this solution only solves the issue of `Tasks` not failing when they don't
+produce `Results` as discussed in [tektoncd/pipeline#3497](https://github.com/tektoncd/pipeline/issues/3497).
+It does not address the use cases for providing default `Results` that can be consumed in subsequent `Tasks`.
+
+
+## Future Work 
+
+Determine if we need default `Results` declared at runtime in the future, and how we can support that.
```

Comment:
can imagine use cases where users could need this at runtime, but the use cases we have defines at at authoring time so agreed with solving for that only in this TEP